src/uu/touch/src/touch.rs

```diff
@@ -248,68 +246,124 @@ fn stat(path: &Path, follow: bool) -> UResult<(FileTime, FileTime)> {
+    let now = time::OffsetDateTime::now_utc();
+
+    let (mut format, mut ts) = match s.chars().count() {
+        15 => (YYYYMMDDHHMM_DOT_SS_FORMAT, s.to_owned()),
+        12 => (YYYYMMDDHHMM_FORMAT, s.to_owned()),
+        13 => (YYMMDDHHMM_DOT_SS_FORMAT, s.to_owned()),
+        10 => (YYMMDDHHMM_FORMAT, s.to_owned()),
+        11 => (YYYYMMDDHHMM_DOT_SS_FORMAT, format!("{}{}", now.year(), s)),
+        8 => (YYYYMMDDHHMM_FORMAT, format!("{}{}", now.year(), s)),
         _ => {
             return Err(USimpleError::new(
                 1,
                 format!("invalid date format {}", s.quote()),
             ))
         }
     };
-
-    let tm = time::strptime(&ts, format)
-        .map_err(|_| USimpleError::new(1, format!("invalid date format {}", s.quote())))?;
-
-    let mut local = to_local(tm);
-    local.tm_isdst = -1;
-    let ft = local_tm_to_filetime(local);
-
-    // We have to check that ft is valid time. Due to daylight saving
-    // time switch, local time can jump from 1:59 AM to 3:00 AM,
-    // in which case any time between 2:00 AM and 2:59 AM is not valid.
-    // Convert back to local time and see if we got the same value back.
-    let ts = time::Timespec {
-        sec: ft.unix_seconds(),
-        nsec: 0,
-    };
-    let tm2 = time::at(ts);
-    if tm.tm_hour != tm2.tm_hour {
-        return Err(USimpleError::new(
-            1,
-            format!("invalid date format {}", s.quote()),
-        ));
+    // workaround time returning Err(TryFromParsed(InsufficientInformation)) for year w/
+    // repr:last_two
+    // https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1ccfac7c07c5d1c7887a11decf0e1996
+    if s.chars().count() == 10 {
+        format = YYYYMMDDHHMM_FORMAT;
+        ts = "20".to_owned() + &ts;
+    } else if s.chars().count() == 13 {
+        format = YYYYMMDDHHMM_DOT_SS_FORMAT;
+        ts = "20".to_owned() + &ts;
+    }
+    if (format == YYYYMMDDHHMM_DOT_SS_FORMAT || format == YYMMDDHHMM_DOT_SS_FORMAT)
+        && ts.ends_with(".60")
+    {
+        // Work around to disable leap seconds
+        ts = ts.replace(".60", ".59");
     }
+    let tm = time::PrimitiveDateTime::parse(&ts, &format)
+        .map_err(|_| USimpleError::new(1, format!("invalid date ts format {}", ts.quote())))?;
+
+    let local = to_local(tm);
+    let ft = local_dt_to_filetime(local);
+
+    // // We have to check that ft is valid time. Due to daylight saving
+    // // time switch, local time can jump from 1:59 AM to 3:00 AM,
+    // // in which case any time between 2:00 AM and 2:59 AM is not valid.
+    // // Convert back to local time and see if we got the same value back.
+    // let ts = time::Timespec {
+    //     sec: ft.unix_seconds(),
+    //     nsec: 0,
+    // };
+    // let tm2 = time::at(ts);
+    // if tm.tm_hour != tm2.tm_hour {
+    //     return Err(USimpleError::new(
+    //         1,
+    //         format!("invalid date format {}", s.quote()),
+    //     ));
+    // }
```

Comment:
Why is this commented out? Is it no longer necessary or is it a remaining thing to port?

Reply:
I missed that, @g-k do you remember ? :)

Reply:
it isn't breaking any test on our side or GNU's
So, I am in favor of remove this piece of code

Reply:
Ugh yeah. This is a remaining thing to port, but also an edge case and probably hard to run into in practice.

IIRC time 0.3 dropped TZ support and removed the Tm struct and [`tm_isdst` flag](https://docs.rs/time/0.1.43/time/struct.Tm.html#structfield.tm_isdst), so I wasn't sure how to proceed since it didn't look like there were good alternatives.

Dropping the code and waiting for support in [icu4x](https://crates.io/crates/icu) seems like a reasonable path to me (icu would also potentially let coreutils handle localized strftime args).


Mentioned it a bit in this part of https://github.com/uutils/coreutils/pull/3319#issue-1182586085 but I had the wrong struct and flag name: 

> How does this project want to handle datetimes and TZ data? This PR disables touch filetime printing daylight savings time tests. time 0.3 removes timezone support and the Timespec DST flag (upstream issue https://github.com/time-rs/time/discussions/351). Unicode icu4x might provide longer term stability and allow us to localize dates and times for [strftime %b and %c](https://man7.org/linux/man-pages/man3/strftime.3.html) currently we assume LC_ALL=C e.g. from https://github.com/unicode-org/icu4x/issues/1317